ngraph/core/include/ngraph/op/ctc_greedy_decoder_seq_len.hpp

```diff
@@ -0,0 +1,80 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v6
+        {
+            /// \brief Operator performing CTCGreedyDecoder
+            ///
+            class NGRAPH_API CTCGreedyDecoderSeqLen : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+                CTCGreedyDecoderSeqLen() = default;
+                /// \brief Constructs a CTCGreedyDecoderSeqLen operation
+                ///
+                /// \param input                Logits on which greedy decoding is performed
+                /// \param seq_len              Sequence lengths
+                /// \param blank_index          Blank index
+                /// \param merge_repeated       Whether to merge repeated labels
+                /// \param classes_index_type   Specifies the output classes_index tensor type
+                /// \param sequence_length_type Specifies the output sequence_length tensor type
+                CTCGreedyDecoderSeqLen(const Output<Node>& input, const Output<Node>& seq_len);
+
+                CTCGreedyDecoderSeqLen(const Output<Node>& input,
+                                       const Output<Node>& seq_len,
+                                       const Output<Node>& blank_index,
+                                       const bool merge_repeated = true,
+                                       const element::Type& classes_index_type = element::i32,
+                                       const element::Type& sequence_length_type = element::i32);
+
+                void validate_and_infer_types() override;
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+                bool get_merge_repeated() const { return m_merge_repeated; }
+                const element::Type& get_classes_index_type() const { return m_classes_index_type; }
+                void set_classes_index_type(const element::Type& classes_index_type)
+                {
+                    m_classes_index_type = classes_index_type;
+                }
+
+                const element::Type& get_sequence_length_type() const
+                {
+                    return m_sequence_length_type;
+                }
+                void set_sequence_length_type(const element::Type& sequence_length_type)
+                {
+                    m_sequence_length_type = sequence_length_type;
+                }
```

Comment:
I am not sure that we need `set` methods because setting attributes through a constructor is sufficient. Please remove it.

Reply:
Sometimes, especially in transformations it is very useful that you can change the attribute value, so you don't need to create a new instance of the operation. So I vote for keeping this function.

But in this particular case we should call the `validate_and_infer_type` after changing the type of the output because otherwise the type of output will be wrong.

@ilyachur , could you comment on that? Will it be enough to just call the `validate_and_infer_type` to make the graph consistent?

Reply:
Yes `validate_and_infer_types()` should be enough.